runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/graph/Networks.java

```diff
@@ -143,8 +146,44 @@ public final NodeT apply(NodeT input) {
     return visitedNodes;
   }
 
-  /** Returns a set of nodes sorted in topological order. */
-  public static <NodeT, EdgeT> Iterable<NodeT> topologicalOrder(Network<NodeT, EdgeT> network) {
+  /**
+   * Returns a set of nodes sorted in topological order.
+   *
+   * <p>The order that unordered nodes appear in the result iterable is arbitrary.
```

Comment:
Its best to just say that we rely on the network to provide an ordering based upon its node order.